sim/main_game.py

```diff
@@ -62,23 +140,25 @@ def _check_events(self):
                     characters = self._get_collided_characters()
                     if characters:
                         character = characters[0]
-                        character.text_box.active = True
+                        text_box_event = character.text_box.box_settings
```

Comment:
this code is quite deeply nested, can we break things out ("extract method refactoring") to simplify?